src/time_utils.c

```diff
@@ -588,3 +591,45 @@ ts_now_mock(PG_FUNCTION_ARGS)
 	res = TimestampTzGetDatum(GetCurrentTimestamp());
 	return res;
 }
+
+Datum
+ts_make_range(PG_FUNCTION_ARGS)
+{
+	Oid rngtypid = get_fn_expr_rettype(fcinfo->flinfo);
+	TypeCacheEntry *typcache = range_get_typcache(fcinfo, rngtypid);
+#if PG16_GE
+	Node *escontext = fcinfo->context;
+#endif
+	RangeBound lower = {
+		.val = PG_ARGISNULL(1) ? 0 : PG_GETARG_DATUM(1),
+		.infinite = PG_ARGISNULL(1),
+		.inclusive = true,
+		.lower = true,
+	};
+	RangeBound upper = {
+		.val = PG_ARGISNULL(2) ? 0 : PG_GETARG_DATUM(2),
+		.infinite = PG_ARGISNULL(2),
+		.inclusive = false,
+		.lower = false,
+	};
+
+	/* Need to check the types of the lower and upper values. They should
+	 * match the returned range. */
+#if PG16_LT
+	PG_RETURN_RANGE_P(make_range(typcache, &lower, &upper, false));
+#else
+	PG_RETURN_RANGE_P(make_range(typcache, &lower, &upper, false, escontext));
+#endif
+}
+
+Datum
+ts_get_min_for_type(PG_FUNCTION_ARGS)
+{
+	PG_RETURN_INT64(ts_time_get_min(PG_GETARG_OID(0)));
+}
+
+Datum
+ts_get_max_for_type(PG_FUNCTION_ARGS)
+{
+	PG_RETURN_INT64(ts_time_get_max(PG_GETARG_OID(0)));
+}
```

Comment:
Thinking if we also should include `ts_get_nobegin_or_min_for_type` and `ts_get_noend_or_max_for_type`.

And about naming IMHO we can get rid of `_for_type` since the parameter is a regtype so the API is very clear to me